src/librustdoc/clean/mod.rs

```diff
@@ -1396,7 +1396,28 @@ fn clean_qpath<'tcx>(hir_ty: &hir::Ty<'tcx>, cx: &mut DocContext<'tcx>) -> Type
                 ty::Projection(proj) => Res::Def(DefKind::Trait, proj.trait_ref(cx.tcx).def_id),
                 // Rustdoc handles `ty::Error`s by turning them into `Type::Infer`s.
                 ty::Error(_) => return Type::Infer,
-                _ => bug!("clean: expected associated type, found `{:?}`", ty),
+                // Otherwise, this is an inherent associated type.
+                _ => {
```

Comment:
Why is this branch required? Why doesn't `hir_ty_to_ty` work?

Reply:
Because apparently there is no way to convert a MIR ty to a cleaned Type in rustdoc. But I might be wrong..

Reply:
That's what `clean_middle_ty` does, see the previous arm `QPath::Resolved`.